source/yats/include/yats/scheduler.h

```diff
@@ -35,20 +37,54 @@ class scheduler
             to_run.push_back(elem.get());
         }
 
-        while (to_run.size() > 0)
-        {
-            auto runnable = std::find_if(to_run.begin(), to_run.end(), [](abstract_task_container* task) {
-                return task->can_run();
-            });
+        std::function<void()> check_runnable;
+
+        check_runnable = [&to_run, &check_runnable, this]() {
+            std::lock_guard<std::mutex> lock(m_mutex);
+
+            while (true)
+            {
+                auto runnable = std::find_if(to_run.begin(), to_run.end(), [](abstract_task_container* task) {
+                    return task->can_run();
+                });
 
-            (*runnable)->run();
+                if (runnable == to_run.end())
+                {
+                    return;
+                }
+                auto task = *runnable;
+                to_run.erase(runnable);
 
-            to_run.erase(runnable);
+                m_threads.emplace_back([&to_run, &check_runnable, task]() {
+                    task->run();
+                    check_runnable();
+                });
+            }
+        };
+
+        check_runnable();
+        while (true)
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
Why is this line required? Looks like a quick fix,

Reply:
It is not perfect but the easiest solution until someone implements #72

Reply:
Ok. Sry. To just get it running it is totally fine.